Articles/Blog/wikize_refs.py

```diff
@@ -0,0 +1,265 @@
+        ol = re.sub("<pus>","<sup>", ol)
+        ol = re.sub("</pus>","</sup>", ol)
+        outlines.append(ol)
+
+    #
+    # Write original set of refs embedded in comments
+    #
+    outlines.append("\n<br>\n\n<!---\n")
+    for l in original_refs:
+        outlines.append(l)
+    outlines.append("\n--->\n<br>\n\n")
+
+    #
+    # Write link definitions with links to anchors in reference table
+    #
+    remapped_ref_map = {v[3]:[v[0],v[1],v[2],k] for k,v in ref_map.items()}
+    for k,v in sorted(remapped_ref_map.items()):
+        outlines.append("[%d]: #ref%d \"%s\"\n"%(k, k, v[0]))
+
+    #
+    # Finally, write the references and off-page links as a table
+    # where each row defines an anchor for one of the link definitions, above
+    #
+    outlines.append("\n<br>\n\n")
+    outlines.append("References | &nbsp;\n")
+    outlines.append(":--- | :---\n")
+    for k,v in sorted(remapped_ref_map.items()):
+        if (not v[0] or v[0].isspace()) and (not v[2] or v[2].isspace()):
+            outlines.append("<a name=\"ref%d\"></a>%d | %s\n"%(k, k, v[1]))
+        elif v[0] in v[2]:
+            outlines.append("<a name=\"ref%d\"></a>%d | [%s](%s)\n"%(k, k, v[2], v[1]))
+        else:
+            outlines.append("<a name=\"ref%d\"></a>%d | [%s %s](%s)\n"%(k, k, v[0], v[2], v[1]))
+
+    #
+    # Write all comment lines
+    #
+    outlines.append("\n<br>\n\n")
+    for l in comment_lines:
+        outlines.append(l)
+    outlines.append("\n")
+
+    #
+    # write warning comment about this being auto-generated
+    #
+    outlines.append("<!--- WARNING: Auto-generated with wikize-refs.py from %s --->\n"%mdfile)
+
+    return outlines
+
+def write_output_temp_file(outlines):
+    with tempfile.NamedTemporaryFile(mode='w', delete=False) as outf:
+        outf.writelines(["%s\n" % item  for item in outlines])
+        return outf.name
+
+def rename_files(in_filename, tmp_filename, out_filename, no_rdonly):
+    if not out_filename:
+        new_in_filename = "%s.src.md"%os.path.splitext(in_filename)[0]
+        if os.path.exists(new_in_filename):
+            print "Cannot rename \"%s\" to \"%s\" because it already exists"%(in_filename, new_in_filename)
+            exit()
+        os.rename(in_filename, new_in_filename)
+        os.rename(tmp_filename, in_filename)
+        if not no_rdonly:
+            os.chmod(in_filename, stat.S_IRUSR | stat.S_IRGRP | stat.S_IROTH)
+    else:
+        if os.path.exists(out_filename):
+            print "Cannot write to \"%s\" because it already exists"%out_filename
+            exit()
+        os.rename(tmp_filename, out_filename)
+        if not no_rdonly:
+            os.chmod(out_filename, stat.S_IRUSR | stat.S_IRGRP | stat.S_IROTH)
+
+def main():
+
+    #
+    # Process command line
+    #
+    opts, mdfile = parse_args()
+    vopts = vars(opts)
```

Comment:
Never saw this usage before.  I always just gave the option an explicit name in the `parser.add_option()` command.  That makes it explicit what the option should be called in Python code and makes it easy to use.